var/spack/repos/builtin/packages/elk/package.py

```diff
@@ -3,130 +3,167 @@
+        src_blis = 'blis_stub.f90'
+        #we will overwrite this if we detect request for 
+        #openblas support. ATM activate stub
+        src_oblas = 'oblas_stub.f90'
+        #we will overwrite this if we detect request for 
+        #libxc support. ATM activate stub
+        src_libxc = 'libxcifc_stub.f90'
+        lib_libxc = '  '
+        #we will overwrite this if we detect request for 
+        #fftw support. ATM activate stub
+        src_fft = 'zfftifc.f90'
+        lib_fft = 'fftlib.a' 
+        #we will overwrite this if we detect request for 
+        #wannier90 support. ATM activate stub
+        src_w90s = 'w90_stub.f90'
+        lib_w90  = '  '
+        #default compiler flags used in elk
+        dfflags  = "-O3 -ffast-math -funroll-loops -fopenmp"
+        #elkFC = mpiCC/FC fi +mpi of CC/FC otherwise
+        #elk uses a stub module in case MPI is absent.
+        if '+mpi' in spec :
+            elk_f77     = spec['mpi'].mpif77
+            elk_f90     = spec['mpi'].mpifc
+            src_mpi     = ' '
+        else :
+            elk_f77     = spack_f77
+            elk_f90     = spack_fc
+            src_mpi     = 'mpi_stub.f90'
+        #elk blas+lapack
+        #elk requires specific stubs for missing openblas and mkl
+        #we do not support linking against mkl yet.
+        if 'linal=openblas' in spec :
+            lib_lpk   = '-lopenblas'
+            src_oblas = '  '
+        if  'fft=fftw' in spec :
+            lib_fft = '-lfftw3'
+            src_fft = 'zfftifc_fftw.f90'
+        if '+libxc' in spec :
+            src_libxc = 'libxcf90.f90 libxcifc.f90'
+            lib_libxc = '-lxcf90 -lxc'
+        if '+wannier90' in spec :
+            src_w90s = '  '
+            lib_w90  = '-lwannier'
+        #write generated config into make.inc
+        with open('make.inc', 'w') as inc :
+            inc.write('{0} = {1} \n'.format('MAKE',      'make' ))
+            inc.write('{0} = {1} \n'.format('F77',       elk_f77))
+            inc.write('{0} = {1} \n'.format('F90',       elk_f90 ))
+            inc.write('{0} = {1} \n'.format('F77_OPTS',  dfflags ))
+            inc.write('{0} = {1} \n'.format('F90_OPTS',  dfflags ))
+            inc.write('{0} = {1} \n'.format('LIB_LPK',   lib_lpk ))
+            inc.write('{0} = {1} \n'.format('LIB_FFT',   lib_fft ))
+            inc.write('{0} = {1} \n'.format('SRC_FFT',   src_fft ))
+            inc.write('{0} = {1} \n'.format('SRC_MPI',   src_mpi ))
+            inc.write('{0} = {1} \n'.format('SRC_MKL',   src_mkl))
+            inc.write('{0} = {1} \n'.format('SRC_OBLAS', src_oblas ))
+            inc.write('{0} = {1} \n'.format('SRC_BLIS',  src_blis ))
+            inc.write('{0} = {1} \n'.format('LIB_libxc', lib_libxc ))
+            inc.write('{0} = {1} \n'.format('SRC_libxc', src_libxc ))
+            inc.write('{0} = {1} \n'.format('SRC_W90S',  src_w90s ))
+            inc.write('{0} = {1} \n'.format('LIB_W90',   lib_w90))
```

Comment:
I prefer the old strategy where we append to a list and loop over the elements of the list. Another good strategy would be to use a dict if you need to re-access a specific element later.

Reply:
@adamjstewart I'm new to python and use of dictionary/loop puzzled me for quite a while. If this is to be a community-supported package file, shouldn't it be kept as primitive as possible?

Reply:
I suppose that's true, although I was using `elk` as an example of this in the docs: https://spack.readthedocs.io/en/latest/build_systems/makefilepackage.html#config-file